File: languages/java/src/main/java/com/bloock/sdk/entity/AesEncrypter.java
package com.bloock.sdk.entity;

import com.bloock.sdk.bridge.proto.RecordOuterClass;
import com.bloock.sdk.bridge.proto.RecordOuterClass.EncryptionAlg;

public class AesEncrypter implements Encrypter {
  EncryptionAlg alg;
  EncrypterArgs args;

  public AesEncrypter(String password) {
    this.alg = EncryptionAlg.A256GCM;
    this.args = new EncrypterArgs(password);
  }

  @Override
  public RecordOuterClass.Encrypter toProto() {
    return RecordOuterClass.Encrypter.newBuilder()
        .setAlg(this.alg)
        .setArgs(this.args.toProto())
        .build();
  }
}

class EncrypterArgs {
  String password;

  EncrypterArgs(String password) {
    this.password = password;
  }

  RecordOuterClass.EncrypterArgs toProto() {
    return RecordOuterClass.EncrypterArgs.newBuilder().setPassword(this.password).build();
  }
}

File: languages/php/Bloock/Entity/AesEncrypter.php
<?php

namespace Bloock\Entity;

class AesEncrypter implements Encrypter
{
    public $alg;
    public $args;

    public function __construct(string $password) {
        $this->alg = \Bloock\EncryptionAlg::A256GCM;
        $this->args = new EncrypterArgs($password);
    }

    public function toProto(): \Bloock\Encrypter {
        $p = new \Bloock\Encrypter();
        $p->setAlg($this->alg);
        $p->setArgs($this->args->toProto());
        return $p;
    }
}